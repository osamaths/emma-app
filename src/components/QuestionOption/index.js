import React from 'react';
import {View, StyleSheet, Text, TouchableOpacity} from 'react-native';
import {colors} from '../../js/sharedStyle';

const QuestionOption = ({option, selected, onSelect}) => {
  const isSelected = selected === option;
  return (
    <TouchableOpacity style={styles.container} onPress={() => onSelect(option)}>
      <View style={[styles.radio, isSelected && {borderColor: colors.primary}]}>
        {isSelected ? <View style={styles.circle} /> : null}
      </View>
      <Text style={[styles.option, isSelected && styles.selected]}>
        {option}
      </Text>
    </TouchableOpacity>
  );
};

export default QuestionOption;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginBottom: 8,
  },
  radio: {
    width: 22,
    height: 22,
    borderRadius: 25,
    borderWidth: 2,
    borderColor: colors.text,
    marginRight: 8,
    justifyContent: 'center',
    alignItems: 'center',
  },
  circle: {
    width: 13,
    height: 13,
    borderRadius: 25,
    backgroundColor: colors.primary,
  },
  option: {
    color: colors.text,
    fontSize: 14,
    fontWeight: '500',
  },
  selected: {
    // color: colors.primary,
  },
});
