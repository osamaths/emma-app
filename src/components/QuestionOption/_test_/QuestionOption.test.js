import React from 'react';
import renderer from 'react-test-renderer';
import QuestionOption from '../index';

test('renders correctly', () => {
  const tree = renderer.create(<QuestionOption />).toJSON();
  expect(tree).toMatchSnapshot();
});
