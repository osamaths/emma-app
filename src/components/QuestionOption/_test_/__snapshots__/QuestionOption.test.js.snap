// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 8,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#102027",
          "borderRadius": 25,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 22,
        },
        Object {
          "borderColor": "#4db6ac",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#4db6ac",
          "borderRadius": 25,
          "height": 13,
          "width": 13,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#102027",
          "fontSize": 14,
          "fontWeight": "500",
        },
        Object {},
      ]
    }
  />
</View>
`;
